src/index.js

```diff
@@ -0,0 +1,18 @@
+import './style';
+import './lib/fix-pmc';
+import App from './components/app';
+
+export default App;
+
+if (typeof window!=='undefined') {
+	addEventListener('click', e => {
+		let { target } = e;
+		do {
+			if (target.nodeName === 'A') {
+				history.pushState(null, null, target.pathname);
+				e.preventDefault();
+				return false;
+			}
+		} while ((target = target.parentNode));
```

Comment:
@jakearchibald says he has a platform method for this `do...while` loop

Reply:
It’s called `element.closest()`

Reply:
removed it anyway.